modules/stochastic_tools/src/reporters/EvaluateSurrogate.C

```diff
@@ -0,0 +1,116 @@
+  params.addParam<MultiMooseEnum>(
+      "response_type",
+      rtypes,
+      "The type of return value expected from the surrogate models, a single entry will use it for "
+      "every model. Warning: not every model is able evaluate every response type.");
+  MultiMooseEnum estd("false=0 true=1", "false");
+  params.addParam<MultiMooseEnum>(
+      "evaluate_std",
+      estd,
+      "Whether or not to evaluate standard deviation associated with each sample, a single entry "
+      "will use it for every model. Warning: not every model can compute standard deviation.");
+  return params;
+}
+
+EvaluateSurrogate::EvaluateSurrogate(const InputParameters & parameters)
+  : StochasticReporter(parameters),
+    SurrogateModelInterface(this),
+    _sampler(getSampler("sampler")),
+    _response_types(getParam<MultiMooseEnum>("response_type"))
+{
+  const auto & model_names = getParam<std::vector<UserObjectName>>("model");
+  _model.reserve(model_names.size());
+  for (const auto & nm : model_names)
+    _model.push_back(&getSurrogateModelByName(nm));
+
+  if (_response_types.size() == 1)
+  {
+    for (unsigned int i = 1; i < _model.size(); ++i)
+      _response_types.push_back(_response_types[0]);
+  }
+  if (_response_types.size() != _model.size())
+    paramError("response_type",
+               "Number of entries must be 1 or equal to the number of entries in 'model'.");
+
+  const auto & estd = getParam<MultiMooseEnum>("evaluate_std");
+  if (estd.size() != 1 && estd.size() != _model.size())
+    paramError("evaluate_std",
+               "Nmber of entries must be 1 or equal to the number of entries in 'model'.");
+  _doing_std.resize(_model.size());
+  for (unsigned int i = 0; i < _model.size(); ++i)
+    _doing_std[i] = estd.size() == 1 ? estd[0] == "true" : estd[i] == "true";
+
+  _real_values.resize(_model.size(), nullptr);
+  _real_std.resize(_model.size(), nullptr);
+  _vector_real_values.resize(_model.size(), nullptr);
+  _vector_real_std.resize(_model.size(), nullptr);
+  for (unsigned int i = 0; i < _model.size(); ++i)
+  {
+    if (_response_types[i] == "real")
+    {
+      _real_values[i] = &declareStochasticReporter<Real>(model_names[i], _sampler);
+      if (_doing_std[i])
+        _real_std[i] = &declareStochasticReporter<Real>(model_names[i] + "_std", _sampler);
+    }
+    else if (_response_types[i] == "vector_real")
+    {
+      _vector_real_values[i] =
+          &declareStochasticReporter<std::vector<Real>>(model_names[i], _sampler);
+      if (_doing_std[i])
+        _vector_real_std[i] =
+            &declareStochasticReporter<std::vector<Real>>(model_names[i] + "_std", _sampler);
+    }
+    else
+      paramError("response_type", "Unknown response type ", _response_types[i]);
+  }
+}
+
+void
+EvaluateSurrogate::execute()
+{
+  // Loop over samples
+  dof_id_type ind = 0;
+  for (dof_id_type p = _sampler.getLocalRowBegin(); p < _sampler.getLocalRowEnd(); ++p)
```

Comment:
Not sure if you knew, but this is a thing in libMesh

```suggestion
  for (const auto p : make_range(_sampler.getLocalRowBegin(), _sampler.getLocalRowEnd()))
```

Reply:
Performance shouldn't change, and I think it's clearer. Take this one as you please. I'm just a fan of const-all-the-things, and this helps.